Automattic/wp-calypso
# Logmein etk module

#### Changes proposed in this Pull Request

Replicate logmein from #53443 for ETK

#### Testing instructions

Either 

- Install on sandbox without building:

```
install-plugin.sh notifications add/etk-logmein
```

Or locally,

- Build & watch etk:
```
cd wp-calypso/apps/editing-toolkit
yarn run dev
```
- Sync changes to sandbox:
```
cd wp-calypso/apps/editing-toolkit
yarn run wpcom-sync
```
- Add a sandbox filter to enable the module: `add_filter( 'wpcom_logmein_enable_etk', '__return_true', 11 );`
- Sandbox a simple site with a mapped domain. (Remember to sandbox the unmapped url for GB/etk).
- Open the editor for the site
- Click, right click, ctrl click, middle click, and touch test the permalink in the sidebar (switching from page to block settings and back re-renders the link for a new test, don't need to page reload)
- Also test the snackbar "View Post" that shows up when you save a post as this required a workaround to get working.

Repeat the tests for non-mapped simple sites, and atomic sites (use an ephemeral site and upload the built plugin), both non-mapped simple sites and atomic sites shouldn't have the logmein param applied to links.

Test support page links as well as site links, support page links should not receive logmein params.

![Screenshot 2021-07-28 at 20-41-44 Edit Post ‹ c3po - Simple Site - Mapped — WordPress](https://user-images.githubusercontent.com/811776/127309314-84dddb10-fcca-41e5-aad6-654cd7ade244.png)

Fixes #54108
